app/controllers/api/providers_controller.rb

```diff
@@ -106,8 +106,49 @@ def resume_resource(type, id, _data)
       action_result(false, "Could not resume Provider - #{err}")
     end
 
+    # Process change_password action for a single resource or a collection of resources
+    def change_password_resource(type, id, data = {})
+      if single_resource?
+        change_password(type, id, data)
+      else
+        change_password_multiple_providers(type, id, data)
+      end
+    end
+
     private
 
+    # Process password change request for a single resource
+    #
+    # @raise [BadRequestError] if no id is passed or some required data is missing
+    #                          or some error occur when try to change password on provider client
+    # @raise [NotFoundError]   if there isn't providers with the id passed
+    def change_password(type, id, data)
+      raise BadRequestError, "Must specify an id for change password of a #{type} resource" unless id
+      validate_change_password_request_data(data)
+      api_action(type, id) do |klass|
+        provider = resource_search(id, type, klass)
+        desc = "Change password requested for Physical Provider #{provider.name}"
+        task_id = provider.change_password_queue(User.current_user.userid, data["current_password"], data["new_password"])
```

Comment:
Do all providers respond to this method? I can't seem to find it in the base class.

Reply:
No, for now, only Lenovo's provider respond to this method. You can see the implementation [here](https://github.com/ManageIQ/manageiq/pull/16895/files#diff-54a07cbbdbeffee51b848dc93f209776R62)

Reply:
This could be problematic - would it result in a 500 if a user targets the wrong provider? How do we reveal through the API what this can be used for?